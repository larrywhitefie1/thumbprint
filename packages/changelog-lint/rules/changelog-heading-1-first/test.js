const test = require('../../test-utils');
const rule = require('./');

const valid = ['# Changelog'];
const invalid = [
    `foo

# Changelog`,
    `> Hi There!

# Changelog`,
];

test({
    rule,
    valid,
    invalid,
});
