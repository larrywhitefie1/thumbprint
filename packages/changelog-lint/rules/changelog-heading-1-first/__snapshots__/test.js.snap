// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`invalid case #0 fails 1`] = `
Object {
  "input": "foo

# Changelog",
  "output": Array [
    Object {
      "column": null,
      "fatal": false,
      "line": null,
      "location": Object {
        "end": Object {
          "column": null,
          "line": null,
        },
        "start": Object {
          "column": null,
          "line": null,
        },
      },
      "reason": "Changelog must start with a top level heading named “# Changelog”",
      "ruleId": "changelog-heading-1-first",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;

exports[`invalid case #1 fails 1`] = `
Object {
  "input": "> Hi There!

# Changelog",
  "output": Array [
    Object {
      "column": null,
      "fatal": false,
      "line": null,
      "location": Object {
        "end": Object {
          "column": null,
          "line": null,
        },
        "start": Object {
          "column": null,
          "line": null,
        },
      },
      "reason": "Changelog must start with a top level heading named “# Changelog”",
      "ruleId": "changelog-heading-1-first",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;
